### PR TITLE
print md to screen if path is NULL

### PR DESCRIPTION
**Pull request by maelle**
Or maybe just head and tail... depending on a parameter?

### CODE REVIEW

**Comment by zkamvar** [comment]
> Or maybe just head and tail... depending on a parameter?

I would prefer to continue to have the `$write()` method return the object, invisibly so that it can be used in a chain: `x$add_md("# title",)$write("file.md")$reset()`

File: R/class-yarn.R [hunk 1]
```diff
@@ -84,15 +84,18 @@ yarn <- R6::R6Class("yarn",
...
       stylesheet_path = system.file("extdata", "xml2md_gfm.xsl", package = "tinkr")){
+
+      output <- to_md(self, path, stylesheet_path)
+
       if (is.null(path)) {
-        stop("Please provide a file path", call. = FALSE)
+        cat(output)
       }
-      to_md(self, path, stylesheet_path)
-      invisible(self)
```
Comment (zkamvar) [inline R/class-yarn.R:93]: This part is important for chaining R6 methods (described in the last paragraph of the "Private Members" section: https://r6.r-lib.org/articles/Introduction.html#private-members)

**Review by zkamvar — CHANGES_REQUESTED** [review]
> Or maybe just head and tail... depending on a parameter?

I would prefer to continue to have the `$write()` method return the object, invisibly so that it can be used in a chain: 

```r
x$add_md("# title")$write("file.md")$reset()
```

I think the methods should be semantically concise and avoid the need for several arguments, so having `$write()` both write a file and output markdown seems to be a bit of misnomer. Instead, I think it would be better to have a `$write()` method and a `$head()` and `$tail()` method.

**Comment by maelle** [comment]
I'm a bit lost, what do I need to change?

**Comment by zkamvar** [comment]
Closing this because we came to a consensus that having `head()`, `tail()` and `show()` methods would be better.

**Pull request closed without merging**